oauth2-server/src/main/java/com/clouway/oauth2/IssueNewTokenActivity.java

```diff
@@ -55,6 +73,40 @@ public Response execute(Client client, Request request, DateTime instant) {
 
     BearerToken accessToken = response.accessToken;
 
-    return new BearerTokenResponse(accessToken.value, accessToken.ttlSeconds(instant), response.refreshToken);
+    String idToken = newIdToken().header(new Header("RS256"))
+            .claims(newClaimSet()
+                    .iss(clientUrl)
+                    .aud(client.id)
+                    .sub(identity.id())
+                    .iat(instant.timestamp())
+                    .exp(accessToken.expirationTimestamp())
+                    .certId(certId)
+                    .customClaims(identity.claims())
+                    .build())
+            .signWith(key)
+            .build();
+
+    return new BearerTokenResponse(accessToken.value, accessToken.ttlSeconds(instant), response.refreshToken, idToken);
+  }
+
+  private Entry<String, Block> randomIdentifierKeyPair() {
```

Comment:
Simplified method but still using entry because I need the Key ID as well so that when the token is verified it will be simpler/easier.